pkg/ccl/crosscluster/logical/logical_replication_job_test.go

```diff
@@ -1101,6 +1235,50 @@ func CreateScatteredTable(t *testing.T, db *sqlutils.SQLRunner, numNodes int, db
 	}, timeout)
 }
 
+func GetPGURLs(
+	t *testing.T, s serverutils.ApplicationLayerInterface, numDBs int,
+) ([]url.URL, func()) {
+	result := []url.URL{}
+	cleanups := []func(){}
+	for i := range numDBs {
+		dbName := string(rune('a' + i))
+		resultURL, cleanup := s.PGUrl(t, serverutils.DBName(dbName))
+		result = append(result, resultURL)
+		cleanups = append(cleanups, cleanup)
+	}
+
+	return result, func() {
+		for _, f := range cleanups {
+			f()
+		}
+	}
+}
+
+func VerifyExpectedRowAllServers(
+	t *testing.T, runners []*sqlutils.SQLRunner, expectedRows [][]string, numDBs int,
+) {
+	for i := range numDBs {
+		dbName := string(rune('a' + i))
+		runners[i].CheckQueryResults(t, fmt.Sprintf("SELECT * from %s.tab", dbName), expectedRows)
+	}
+}
+
+func WaitUntilReplicatedTimeAllServers(
```

Comment:
i feel like you could make this function local to the 4 way replication test. This would reduce signature bloat, and this func seems pretty specialized.

Reply:
done